subprojects/base-services/src/test/groovy/org/gradle/api/JavaVersionSpec.groovy

```diff
@@ -98,10 +99,6 @@ public class JavaVersionSpec extends Specification {
         conversionFails("1.54");
         conversionFails("2.0");
         conversionFails("1_4");
-        conversionFails("8.1");
-        conversionFails("9.1");
-        conversionFails("9.0.0");
-        conversionFails("10.1.2");
```

Comment:
Shouldn't we keep the failure for version 10?

Reply:
Well, I'm still hesitating, we don't know what next JDK 9 minor release looks like, not to mention `10`. I understand keeping this failure for safety, but should we make [`JavaVersion.toVersion(10) == JavaVersion.VERSION_1_10`](https://github.com/gradle/gradle/blob/c1fe5e40a76b79a98e8916c2ce3b4e1e6ed3f343/subprojects/base-services/src/test/groovy/org/gradle/api/JavaVersionSpec.groovy#L61) fail as well?

Reply:
The next version was `10` for some time, this is why @melix added the enum in the first place. I think the JavaFX team was using Gradle for their build or something along these lines. So we need to support Java 10 already.

Reply:
That's correct, we need to keep recognizing 10.

Reply:
Okay, I guess I need to move `10.1.2` to `convertsStringToVersion()` method.